diff -Nur '--exclude=.git' '--exclude=tuxtxt' enigma2-nightly.org/lib/python/Screens/InfoBarGenerics.py enigma2-nightly.patched/lib/python/Screens/InfoBarGenerics.py
--- enigma2-nightly.org/lib/python/Screens/InfoBarGenerics.py	2012-03-20 01:26:26.547115310 -0700
+++ enigma2-nightly.patched/lib/python/Screens/InfoBarGenerics.py	2012-03-20 01:26:30.307115296 -0700
@@ -432,8 +432,13 @@
 				"historyBack": (self.historyBack, _("previous channel in history")),
 				"historyNext": (self.historyNext, _("next channel in history")),
 				"openServiceList": (self.openServiceList, _("open servicelist")),
+				"openServiceListFAV": (self.openServiceListFAV, _("open servicelist")),
 			})
 
+	def openServiceListFAV(self):
+		self.servicelist.showFavourites()
+		self.session.execDialog(self.servicelist)
+
 	def showTvChannelList(self, zap=False):
 		self.servicelist.setModeTv()
 		if zap:
diff -Nur '--exclude=.git' '--exclude=tuxtxt' enigma2-nightly.org/lib/driver/vfd.cpp enigma2-nightly.patched/lib/driver/vfd.cpp
--- enigma2-nightly.org/lib/driver/vfd.cpp	2012-03-20 01:26:26.547115310 -0700
+++ enigma2-nightly.patched/lib/driver/vfd.cpp	2012-03-20 01:26:30.307115296 -0700
@@ -25,7 +25,7 @@
 #define VFDDISPLAYCLR		0xc0425b00
 
 bool startloop_running = false;
-static bool icon_onoff[32];
+static bool icon_onoff[45];
 static pthread_t thread_start_loop = 0;
 void * start_loop (void *arg);
 bool blocked = false;
@@ -124,7 +124,7 @@
 	//ioctl(fpc, FRONTPANELSCROLLMODE, &scrollMode);
 
 	// display string
-	char str[] = "Open AR-P ENIGMA2";
+	char str[] = "TESLANET";
 	int length = strlen(str);
 	char dispData[MAX_CHARS + 1];
 	int offset = 0;
@@ -188,16 +190,14 @@
 	evfd vfd;
 	blocked = true;
 	//vfd.vfd_clear_icons();
-	vfd.vfd_write_string("Open AR-P ENIGMA2", true);
+	vfd.vfd_write_string("TESLANET", true);
 	//run 2 times through all icons 
-	for  (int vloop = 0; vloop < 128; vloop++) {
 #if !defined(PLATFORM_FORTIS_HDBOX) && !defined(PLATFORM_OCTAGON1008) && !defined(PLATFORM_ATEVIO7500) && !defined(PLATFORM_CUBEREVO) && !defined(PLATFORM_CUBEREVO_MINI) && !defined(PLATFORM_CUBEREVO_MINI2) && !defined(PLATFORM_CUBEREVO_MINI_FTA) && !defined(PLATFORM_CUBEREVO_250HD) && !defined(PLATFORM_CUBEREVO_2000HD) && !defined(PLATFORM_CUBEREVO_9500HD) && !defined(PLATFORM_HS7810A) && !defined(PLATFORM_SPARK7162)
-		if (vloop%2 == 1) {
-			vfd.vfd_set_icon( (tvfd_icon) (((vloop%32)/2)%16), ICON_OFF, true);
-			//usleep(1000);
-			vfd.vfd_set_icon( (tvfd_icon) ((((vloop%32)/2)%16)+1), ICON_ON, true);
+	for  (int vloop = 0x00; vloop <= 0x2d; vloop++) {
+		vfd.vfd_set_icon((tvfd_icon)vloop , ICON_ON, true);
-		}
+	}
 #else
+	for  (int vloop = 0; vloop < 128; vloop++) {
 		if (vloop%14 == 0 )
 			vfd.vfd_set_brightness(1);
 		else if (vloop%14 == 1 )
@@ -232,13 +230,13 @@
 			vfd.vfd_set_brightness(1);
 		else if (vloop%14 == 13 )
 			vfd.vfd_set_brightness(0);
-#endif		
 		usleep(75000);
 	}
+#endif
 	vfd.vfd_set_brightness(7);
 #if !defined(PLATFORM_FORTIS_HDBOX) && !defined(PLATFORM_OCTAGON1008) && !defined(PLATFORM_ATEVIO7500) && !defined(PLATFORM_CUBEREVO) && !defined(PLATFORM_CUBEREVO_MINI) && !defined(PLATFORM_CUBEREVO_MINI2) && !defined(PLATFORM_CUBEREVO_MINI_FTA) && !defined(PLATFORM_CUBEREVO_250HD) && !defined(PLATFORM_CUBEREVO_2000HD) && !defined(PLATFORM_CUBEREVO_9500HD) && !defined(PLATFORM_HS7810A) && !defined(PLATFORM_SPARK7162)
 	//set all blocked icons
-	for (int id = 0x10; id < 0x20; id++) {
+	for (int id = 0x01; id <= 0x2d; id++) {
 		vfd.vfd_set_icon((tvfd_icon)id, icon_onoff[id]);
 	}
 #endif
@@ -445,7 +443,7 @@
 
 void evfd::vfd_clear_icons()
 {
-	for (int id = 0x10; id < 0x20; id++) {
+	for (int id = 0x01; id <= 0x2d; id++) {
 		vfd_set_icon((tvfd_icon)id, false);
 	}
 	return;
